controllers/operatorscaler_controller.go

```diff
@@ -177,6 +185,57 @@ func (r *OperatorScalerReconciler) isOdfDependenciesCsvReady() error {
 	return nil
 }
 
+func (r *OperatorScalerReconciler) removeStorageSystemOwnerRef() error {
+	r.log.Info("entering removeStorageSystemOwnerRef")
+
+	var returnErr error
+
+	for i := range ResourceMappingList {
+		resMap := &ResourceMappingList[i]
+
+		if resMap.RemoveOwnerRef {
+
+			crList := &metav1.PartialObjectMetadataList{}
+			crList.TypeMeta.APIVersion = resMap.ApiVersion
+			crList.TypeMeta.Kind = resMap.Kind
+
+			if err := r.Client.List(r.ctx, crList); err != nil {
+				if !meta.IsNoMatchError(err) {
+					msg := fmt.Sprintf("failed listing %s", resMap.Kind)
+					r.log.Error(err, msg)
+					multierr.AppendInto(&returnErr, err)
+				}
+			} else {
+				for j := range crList.Items {
+					crItem := &crList.Items[j]
+
+					ownRefLen := len(crItem.OwnerReferences)
+
+					// remove the storagesystem owner reference from the CR
+					crItem.OwnerReferences = slices.DeleteFunc(crItem.OwnerReferences, func(oRef metav1.OwnerReference) bool {
+						return oRef.Kind == "StorageSystem"
+					})
+
+					if len(crItem.OwnerReferences) < ownRefLen {
+						r.log.Info("removing owner reference", "kind", resMap.Kind, "name", crItem.Name)
+
+						if err := r.Client.Update(r.ctx, crItem); err != nil {
```

Comment:
is update allowed on partial object? an update is a full view of the resource and partial object lacks the spec making the stanza except meta be removed at kube end?

Reply:
I need to test this one, I have not tested this code yet. If it does not work I will try patch.

Reply:
I tested it, It was not working, update the code to use patch.

I was getting this error `cannot update using only metadata -- did you mean to patch?`